pkg/liquidity-mining/contracts/admin/GaugeAdder.sol

```diff
@@ -43,143 +48,136 @@ contract GaugeAdder is IGaugeAdder, SingletonAuthentication, ReentrancyGuard {
-
-    /**
-     * @notice Adds a new gauge to the GaugeController for the "Ethereum" type.
-     */
-    function addEthereumGauge(IStakingLiquidityGauge gauge) external override authenticate {
-        // Each gauge factory prevents deploying multiple gauges for the same Balancer pool
-        // however two separate factories can each deploy their own gauge for the same pool.
-        // We then check here to see if the new gauge's pool already has a gauge on the Gauge Controller.
-        IERC20 pool = gauge.lp_token();
-        require(pool != _balWethBpt, "Cannot add gauge for 80/20 BAL-WETH BPT");
-
-        _addGauge(address(gauge), GaugeType.Ethereum);
+    /// @inheritdoc IGaugeAdder
+    function getFactoryForGaugeType(string memory gaugeType)
+        public
+        view
+        override
+        withValidGaugeType(gaugeType)
+        returns (ILiquidityGaugeFactory)
+    {
+        return _gaugeTypeFactory[gaugeType];
     }
 
-    /**
-     * @notice Adds a new gauge to the GaugeController for the "Polygon" type.
-     * This function must be called with the address of the *root* gauge which is deployed on Ethereum mainnet.
-     * It should not be called with the address of the gauge which is deployed on Polygon
-     */
-    function addPolygonGauge(address rootGauge) external override authenticate {
-        _addGauge(rootGauge, GaugeType.Polygon);
+    /// @inheritdoc IGaugeAdder
+    function isGaugeFromValidFactory(address gauge, string memory gaugeType)
+        external
+        view
+        override
+        withValidGaugeType(gaugeType)
+        returns (bool)
+    {
+        return _isGaugeFromValidFactory(gauge, gaugeType);
     }
 
-    /**
-     * @notice Adds a new gauge to the GaugeController for the "Arbitrum" type.
-     * This function must be called with the address of the *root* gauge which is deployed on Ethereum mainnet.
-     * It should not be called with the address of the gauge which is deployed on Arbitrum
-     */
-    function addArbitrumGauge(address rootGauge) external override authenticate {
-        _addGauge(rootGauge, GaugeType.Arbitrum);
-    }
+    // Admin Functions
 
-    /**
-     * @notice Adds a new gauge to the GaugeController for the "Optimism" type.
-     * This function must be called with the address of the *root* gauge which is deployed on Ethereum mainnet.
-     * It should not be called with the address of the gauge which is deployed on Optimism.
-     */
-    function addOptimismGauge(address rootGauge) external override authenticate {
-        _addGauge(rootGauge, GaugeType.Optimism);
-    }
+    /// @inheritdoc IGaugeAdder
+    function addGaugeType(string memory gaugeType) external override authenticate {
+        uint256 gaugeTypeLength = bytes(gaugeType).length;
+        require(
+            gaugeTypeLength > 0 && gaugeTypeLength <= 32,
+            "Input string should be between 1 and 32 characters long"
+        );
```

Comment:
```suggestion
```

We no longer need this

Reply:
I left the non-empty check, which I think makes sense.